File: in_Python/1365 How Many Numbers Are Smaller Than the Current Number.py
class Solution(object):
    def smallerNumbersThanCurrent(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = nums
        L = sorted(nums)
        for i in range(len(L)):
            res[i] = L.index(nums[i])        
        return res
        

File: in_Java/1365 How Many Numbers Are Smaller Than the Current Number.java
class Solution {
    public int[] smallerNumbersThanCurrent(int[] nums) {
        int[] res = new int[nums.length];
        for(int i = 0; i < nums.length; i++) res[i] = nums[i];
        Arrays.sort(nums);
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int cur = nums[i];
            if(!hm.containsKey(cur)){
                hm.put(cur,i);
            }
        }
        for(int i = 0; i < nums.length; i++){
            res[i] = hm.get(res[i]);
        }
        return res;
    }
}